src/massaction_rates.jl

```diff
@@ -3,11 +3,10 @@
 # stochiometric coefficient.
 ###############################################################################
 
-@fastmath function evalrxrate(speciesvec::AbstractVector{T}, rateconst,
-                              stochmat::AbstractVector{Pair{S,V}})::typeof(rateconst) where {T,S,V}
+@inline @fastmath function evalrxrate(speciesvec::AbstractVector{T}, rxidx::S,
```

Comment:
is the inlining necessary here?

Reply:
There's something going on with the compiler that I can't quite figure out. Without the inlining, the previous version, which indexes into the `MassActionJump` in the aggregator and then passes in the stoichiometry and rate, was noticeably faster (10-20%). Inlining eliminated the performance difference. In contrast, I don't think inlining the old version had much effect on performance. I'm not sure why this is -- I haven't quite mastered reading `@code_warntype` yet.

The conclusion though is that the inlining does make a big difference for this version of the function. Since this routine is often the rate limiting part of any Direct method simulation it seemed worth it. (That's why I'm also playing around with static vectors for the internal jump representation -- every bit of performance gain here will help.)

Reply:
Sounds good.